compactor2/src/components/partitions_source/mod.rs

```diff
@@ -0,0 +1,20 @@
+use std::fmt::{Debug, Display};
+
+use async_trait::async_trait;
+use data_types::PartitionId;
+
+pub mod catalog;
+pub mod logging;
+pub mod mock;
+pub mod randomize_order;
+
+//// Source for [partitions](PartitionId).
+#[async_trait]
+pub trait PartitionsSource: Debug + Display + Send + Sync {
```

Comment:
Am I right in thinking:

```suggestion
//// A source of [partitions](PartitionId) that may potentially need compacting.
#[async_trait]
pub trait PartitionsSource: Debug + Display + Send + Sync {
```

Will there need to be some sort of compaction level information passed out of the source, to support L1 compaction etc?

Reply:
> Will there need to be some sort of compaction level information passed out of the source, to support L1 compaction etc?

Not yet, but I guess we need some more information flow through the pipe at one point. I will defer the design of that til we need it though.

I also expect that at one point we need some "annotators" for the different partitions, i.e. a small functions that can add tags / annotations to a partition like "hot/cold" or "high prio / low prio". However this also needs careful design so it fits into the component framework.